plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -162,6 +163,7 @@ public class SemiTransactionalHiveMetastore
             .withDelay(java.time.Duration.ofSeconds(1))
             .withMaxDuration(java.time.Duration.ofSeconds(30))
             .withMaxAttempts(3)
+            .abortOn(UnrecoverableIOException.class)
```

Comment:
Can we always use the `isUnrecoverableIOException` method for consistency?
```suggestion
            .abortOn(UnrecoverableIOException::isUnrecoverableIOException)
```